### PR TITLE
Newtor foreground services

### PR DESCRIPTION
**Pull request by djkazic**
Not yet ready for merge, but close.

### CODE REVIEW

File: android/app/src/main/java/com/blixtwallet/LndMobileScheduledSyncWorker.java [hunk 1]
```diff
@@ -85,8 +87,8 @@ public ListenableFuture<Result> startWork() {
       writeLastScheduledSyncAttemptToDb();
 
       HyperLog.i(TAG, "MainActivity.started = " + MainActivity.started);
-      if (MainActivity.started) {
-        HyperLog.i(TAG, "MainActivity is started, quitting job");
+      if (persistentServicesEnabled || MainActivity.started) {
+        HyperLog.i(TAG, "MainActivity is started, persistentServicesEnabled = " + persistentServicesEnabled + ", quitting job");
```
Comment (hsjoberg) [inline android/app/src/main/java/com/blixtwallet/LndMobileScheduledSyncWorker.java:91]: nit

```suggestion
        HyperLog.i(TAG, "MainActivity is started or persistentServicesEnabled = " + persistentServicesEnabled + ", quitting job");
```

File: android/app/src/main/java/com/blixtwallet/LndMobileService.java [hunk 2]
```diff
@@ -28,12 +33,17 @@
...
+import com.facebook.react.modules.storage.ReactDatabaseSupplier;
+
 import com.google.protobuf.ByteString;
 
 import com.hypertrack.hyperlog.HyperLog;
 
 public class LndMobileService extends Service {
   private static final String TAG = "LndMobileService";
+  private final int ONGOING_NOTIFICATION_ID = 1;
+  private final String CHANNEL_DEFAULT_IMPORTANCE = "blixt";
```
Comment (hsjoberg) [inline android/app/src/main/java/com/blixtwallet/LndMobileService.java:46]: I don't think this is one is used anywhere

**Review by hsjoberg — CHANGES_REQUESTED** [review]

File: src/windows/Settings/Settings.tsx [hunk 3]
```diff
@@ -597,6 +597,10 @@ ${t("LN.inbound.dialog.msg3")}`
           text: t("buttons.yes",{ns:namespaces.common}),
           onPress: async () => {
             try {
+              const torEnabled = useStoreState((store) => store.settings.torEnabled);
```
[the remark this answers is not in the document]
Comment (djkazic) [reply src/windows/Settings/Settings.tsx]: I have made the call to `NativeModules.BlixtTor.stopTor()` safe to call if Tor is not running. This should work around the need to get `torEnabled` here.

File: android/app/src/main/java/com/blixtwallet/LndMobileService.java (hunk 2 above)
Comment (djkazic) [reply android/app/src/main/java/com/blixtwallet/LndMobileService.java]: Fixed.

File: android/app/src/main/java/com/blixtwallet/LndMobileScheduledSyncWorker.java (hunk 1 above)
Comment (djkazic) [reply android/app/src/main/java/com/blixtwallet/LndMobileScheduledSyncWorker.java]: Nit fixed.

File: android/app/src/main/java/com/blixtwallet/BlixtTor.java [hunk 4]
```diff
@@ -16,39 +23,85 @@
...
     public void onServiceDisconnected(ComponentName arg0) {
       Log.i(TAG, "onServiceDisconnected");
     }
+
+    public Notification getNotification(Context context) {
+      Intent notificationIntent = new Intent (context, MainActivity.class);
+      PendingIntent pendingIntent =
+        PendingIntent.getActivity(context, 0, notificationIntent, PendingIntent.FLAG_IMMUTABLE);
+      if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
+        NotificationChannel channel = new NotificationChannel("com.blixtwallet", "blixt", NotificationManager.IMPORTANCE_NONE);
```
[the remark this answers is not in the document]
Comment (djkazic) [reply android/app/src/main/java/com/blixtwallet/BlixtTor.java]: Set `BuildConfig.APPLICATION_ID` in place of `com.blixtwallet`.

[the remark this answers is not in the document]
Comment (djkazic) [reply android/app/src/main/java/com/blixtwallet/BlixtTor.java]: Fixed this in `LndMobileService` as well.

**Comment by hsjoberg** [comment]
Work continues in #1188 and `newtor-and-persistent` branch.

**Pull request closed without merging**